docs/source/start/declarative.rst

```diff
@@ -0,0 +1,148 @@
+..
+   SPDX-FileCopyrightText: Copyright DB Netz AG and the capellambse contributors
+   SPDX-License-Identifier: Apache-2.0
+
+.. _declarative-modelling:
+
+*********************
+Declarative modelling
+*********************
+
+Since v0.5, py-capellambse supports declarative modelling with the
+:py:mod:`capellambse.decl` module. This requires the optional dependency
+``capellambse[decl]`` to be installed.
+
+The YAML-based declarative modelling engine combines a few simple concepts into
+a powerful, but easy to use file format. These files can then be applied to any
+model supported by py-capellambse.
+
+Example
+=======
+
+Here is an example YAML file that declares a simple coffee machine with a
+couple of functions, and functional exchanges between them:
+
+.. literalinclude:: ../../../tests/data/decl/coffee-machine.yml
+   :language: yaml
+   :lines: 4-
+   :lineno-start: 1
+   :linenos:
+
+CLI usage
+---------
+
+If the additional optional dependency ``capellambse[decl,cli]`` is installed,
+this file can be applied from the command line. Assuming it is saved as
+``coffee-machine.yml``, it can then be applied to a locally stored Capella
+model like this:
+
+.. code-block:: sh
+
+   python -m capellambse.decl --model path/to/model.aird coffee-machine.yml
+
+Refer to the :py:func:`capellambse.cli_helpers.loadcli` documentation to find
+out the supported argument format for ``--model``.
+
+API usage
+---------
+
+Declarative YAML can also be applied programmatically, by calling the
+:py:func:`capellambse.decl.apply` function. It takes a (loaded) CapellaMBSE
+model, and either a path to a file or a file-like object. To pass in a string
+containing YAML, wrap it in ``io.StringIO``:
+
+.. code-block:: python
+   :emphasize-lines: 5
+
+   import io, capellambse.decl
+   my_model = capellambse.MelodyModel(...)
+   my_yaml = "..."
+
+   capellambse.decl.apply(my_model, io.StringIO(my_yaml))
+
+   my_model.save()
+
+Format description
+==================
+
+The expected YAML follows a simple format, where a parent object (i.e. an
+object that already exists in the model) is selected, and one or more of three
+different operations is applied to it:
+
+- ``create``-ing new child objects,
+- ``modify``-ing the object itself, or
+- ``delete``-ing one or more children.
```

Comment:
What do you think about documenting the modify and delete operation also? IMO `create` is covered very well and indepth, but we leave room for ✨creativity✨ of users... and as you know users can be very ✨creative✨.

In a future PR we could add other operations here like `extend`, etc.

Reply:
Good point. Not sure how I managed to miss that. I'm gonna write up something more about it.